src/representation_theory/monoid_algebra_basis.lean

```diff
@@ -0,0 +1,213 @@
+/-
+Copyright (c) 2022 Amelia Livingston. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Amelia Livingston
+-/
+
+import representation_theory.Rep
+import representation_theory.basic
+
+/-!
+# The structure of the `k[G]`-module `k[Gⁿ]`
+
+This file contains API for the `k[G]`-module `k[Gⁿ]`, where `k` is a commutative ring, `G` is
+(mainly) a group, and the module structure is induced by the diagonal action of `G` on `Gⁿ.`
+
+In particular, we define `k[G]`-linear maps between `k[Gⁿ⁺¹]` and `k[G] ⊗ₖ k[Gⁿ]` which we will
+later show define an isomorphism. From this we will deduce that `k[Gⁿ⁺¹]` is free over `k[G]`.
+
+The freeness of these modules means we can use them to build a projective resolution of the
+(trivial) `k[G]`-module `k`, which is useful in group cohomology.
+
+## Main definitions
+
+  * mul_action_to_ρ
+  * mul_action_to_Rep
+  * to_tensor
+  * of_tensor
+
+## Implementation notes
+
+We bundle the type `k[Gⁿ]` - or more generally `k[H]` when `H` has `G`-action - and the
+`k[G]`-module structure together as a term of type `Rep k G`, and we call it
+`mul_action_to_Rep k G H.` The representation `ρ: G →* End(k[H])` is then induced by the `G`-action
+on `H`. This is so we only define a `k[G]`-module instance on `mul_action_to_Rep k G H` and not
+directly on `monoid_algebra k H`, which could cause diamonds.
+
+-/
+
+noncomputable theory
+universes v u
+variables {k G : Type u} [comm_ring k]
+
+open monoid_algebra
+namespace Rep
+
+section
+variables [monoid G]
+
+/- Not sure where to put the next several things; either `representation_theory.Rep` or
+`representation_theory.basic`, neither of which import the other. -/
+
+/-- A term of type `Rep k G` has the natural structure of a `k[G]`-module, with an element
+`g : G` acting by `g • x = (ρ g) x` -/
+instance module' (V : Rep k G) : module (monoid_algebra k G) V :=
+representation.as_module V.ρ
```

Comment:
I thought we decided this wasn't going to be an instance, and we would work directly with `\rho`?

Reply:
Okay, I've been using the module instance so I can ultimately use `module.free` in the next PR, but I guess that's not strictly necessary. I've refactored to just use `\rho` & now I'm figuring out how I should change the documentation accordingly.